pina/pinn.py

```diff
@@ -121,6 +134,39 @@ def load_state(self, filename):
 
         return self
 
+    def _create_dataloader(self):
+        """Private method for creating dataloader
+
+        :return: dataloader
+        :rtype: torch.utils.data.DataLoader
+        """
+        if self.batch_size is None:
+            return [self.input_pts]
+
+        def custom_collate(batch):
+            # extracting pts labels
+            _, pts = list(batch[0].items())[0]
+            labels = pts.labels
+            # calling default torch collate
+            collate_res = default_collate(batch)
+            # save collate result in dict
+            res = {}
+            for key, val in collate_res.items():
+                val.labels = labels
+                res[key] = val
+            return res
+
+        # creating dataset, list of dataset for each location
+        datasets = [MyDataSet(key, val)
+                    for key, val in self.input_pts.items()]
+        # creating dataloader
+        dataloaders = [DataLoader(dataset=dat,
+                                  batch_size=self.batch_size,
+                                  collate_fn=custom_collate)
+                       for dat in datasets]
+
+        return dict(zip(self.input_pts.keys(), dataloaders))
```

Comment:
What if we encapsulate these lines inside the DataSet class?
Something in order to have `for batch in dataset.dataloader` in the training loop.

Reply:
So I believe that creating a DataLoader Class is the best option, do you think it should be put in the utils file?

Reply:
solved in commit [b9cb585](https://github.com/mathLab/PINA/pull/51/commits/b9cb585e0ab2ca4598a5e05ae7c2241ce3e4beb1)